File: app/src/main/java/fr/frogdevelopment/nihongo/dico/data/rest/EntryDetails.java
package fr.frogdevelopment.nihongo.dico.data.rest;

import java.io.Serializable;
import java.util.Set;

public class EntryDetails implements Serializable {

    private static final long serialVersionUID = 1L;

    public int entrySeq;
    public String kanji;
    public String kana;
    public String reading;
    public Set<String> pos;
    public Set<String> field;
    public Set<String> misc;
    public String info;
    public Set<String> dial;
    public String gloss;
}

File: app/src/main/java/fr/frogdevelopment/nihongo/dico/data/rest/EntryDetails.kt
package fr.frogdevelopment.nihongo.dico.data.rest

import java.io.Serializable

class EntryDetails(
        val entrySeq: Int?,
        val kanji: String?,
        val kana: String,
        val reading: String,
        val pos: Set<String>?,
        val field: Set<String>?,
        val misc: Set<String>?,
        val info: String?,
        val dial: Set<String>?,
        val gloss: String? = null
) : Serializable